File: test/merkletreejs/merkle_proof/merkle_proof_test.js
const { MerkleTree } = require('merkletreejs');
const SHA256 = require('crypto-js/sha256');

const leaves = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'].map(x => Buffer.from(x, 'utf-8')).map(SHA256);
const tree = new MerkleTree(leaves, SHA256);
const leaf = SHA256(Buffer.from('a', 'utf-8'));
const proof = tree.getProof(leaf).map(node => ({ data: node.data.toString('hex'), position: node.position }));

console.log(JSON.stringify({ proof, isValid: tree.verify(proof, leaf, tree.getRoot()) }))

File: test/merkletreejs/merkle_proof/merkle_proof_test.py
from merkly.mtree import MerkleTree
import hashlib
import json


def sha256(x, y):
    data = x + y
    return hashlib.sha256(data).digest()


leaves = ["a", "b", "c", "d", "e", "f", "g", "h"]
leaves = list(map(lambda x: x.encode(),leaves))
tree = MerkleTree(leaves, sha256)
leaf = ("a").encode()
proof = tree.proof(leaf)
formatted_proof = [
    {"data": node.data.hex(), "position": node.side.name.lower()} for node in proof
]

print(json.dumps({"proof": formatted_proof, "isValid": tree.verify(proof, leaf)}))
